template/src/pdk.ts.ejs

```diff
@@ -8,21 +8,45 @@ function isNull(v: any): boolean {
 
 function cast(caster: (v: any) => any, v: any): any {
   if (isNull(v)) return v
-  if (Array.isArray(v)) return v.map(caster)
   return caster(v)
 }
 
-function dateToJson(v: Date): string {
+function castArray(caster: (v: any) => any) {
+  return (v?: Array<any>) => {
+    if (isNull(v)) return v
```

Comment:
A (kind-of lengthy) question! So, IIUC right now the JS bindgen is more permissive than other bindgens about what values it accepts. That can lead to some surprise down the line for hosts if they build their host service and initial plugins using JS. (This happened in extendabot: some interfaces were actually nullable but not marked as such. The JS bindgen plugins worked, but go, rust, and others failed in ways that were difficult to debug from the guest side.)

Should we track type nullability here and fail if an array (or array member) is set to `null` inappropriately to match the other bindgens? (It's possible we're doing that and I just missed it, too!)

Reply:
This kind of comes at the question of the line b/w types and validation. Who is responsible for validation and where should that happen? I wanted this casting code to be agnostic to that. Though I understand with other, typed languages you get more validation out of the box. I still need to think a bit about where they should happen. We could maybe follow up with a stricter casting routine. I'm still holding out hope that we can use a dependency for all this code and delete it.

Reply:
Something to note about the casting code if it's not clear, we don't cast the whole object. only the parts that need it. So i do think if we consider some kind of validation we may need to do that on the whole object somehow.